pkg/operator/resource/resourceapply/core.go

```diff
@@ -280,14 +280,23 @@ func ApplySecret(client coreclientv1.SecretsGetter, recorder events.Recorder, re
 	if klog.V(4) {
 		klog.Infof("Secret %s/%s changes: %v", required.Namespace, required.Name, JSONPatchSecretNoError(existing, existingCopy))
 	}
-	actual, err := client.Secrets(required.Namespace).Update(context.TODO(), existingCopy, metav1.UpdateOptions{})
-	reportUpdateEvent(recorder, existingCopy, err)
 
```

Comment:
You need to properly default empty with what the the server does.

Reply:
added the defaulting